client/rest-high-level/src/main/java/org/elasticsearch/client/Validatable.java

```diff
@@ -18,24 +18,19 @@
  */
 package org.elasticsearch.client;
 
+import java.util.Optional;
+
 /**
  * Defines a validation layer for Requests.
  */
 public interface Validatable {
-    ValidationException EMPTY_VALIDATION = new ValidationException() {
-        @Override
-        public void addValidationError(String error) {
-            throw new UnsupportedOperationException("Validation messages should not be added to the empty validation");
-        }
-    };
-
     /**
-     * Perform validation. This method does not have to be overridden in the event that no validation needs to be done.
+     * Perform validation. This method does not have to be overridden in the event that no validation needs to be done,
+     * or the validation was done during object construction time.
      *
-     * @return potentially null, in the event of older actions, an empty {@link ValidationException} in newer actions, or finally a
-     * {@link ValidationException} that contains a list of all failed validation.
+     * @return An {@link Optional} {@link ValidationException} that may or may not contain a list of validation errors.
```

Comment:
If we aren't checking for `isEmpty()`, we should make sure to say in the documentation that if the Optional isn't empty, the ValidationException should contain at least one error, rather than "may or may not".

Reply:
yes true. Anything non null should be thrown here.